src/compiler/checker.ts

```diff
@@ -12829,7 +12829,7 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
         const baseTypes = getBaseTypes(source);
         if (baseTypes.length) {
             if (source.symbol && members === getMembersOfSymbol(source.symbol)) {
-                members = createSymbolTable(source.declaredProperties);
+                members = createSymbolTable(members.values(), SymbolFlags.TypeParameter);

```

Comment:
arguably this could be changed to smth like:
```suggestion
                members = createSymbolTable(source.declaredProperties);
                const indexSymbol = members.get(InternalSymbolName.Index);
                if (indexSymbol) {
                    members.set(InternalSymbolName.Index, indexSymbol);
                }
```

but i like the current version better (if only there is no issue with it rn). It's easy to forget forwarding some of those special members and the current version ensures that all of them **are** forwarded.

Reply:
I'm definitely liking the change less now that `createSymbolTable` is doing filtering; I think I'd prefer `createSymbolTable` not changing at all and instead writing the explicit code with a comment if needed, which I think is must more common.

Reply:
done